src/NewsService.Business/CreateNewsCommand.cs

```diff
@@ -1,102 +1,131 @@
-﻿using LT.DigitalOffice.Kernel.AccessValidatorEngine.Interfaces;
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using LT.DigitalOffice.Kernel.AccessValidatorEngine.Interfaces;
 using LT.DigitalOffice.Kernel.Broker;
 using LT.DigitalOffice.Kernel.Constants;
 using LT.DigitalOffice.Kernel.Enums;
-using LT.DigitalOffice.Kernel.Exceptions.Models;
 using LT.DigitalOffice.Kernel.FluentValidationExtensions;
 using LT.DigitalOffice.Kernel.Responses;
 using LT.DigitalOffice.Models.Broker.Common;
 using LT.DigitalOffice.NewsService.Business.Interfaces;
 using LT.DigitalOffice.NewsService.Data.Interfaces;
 using LT.DigitalOffice.NewsService.Mappers.Models.Interfaces;
 using LT.DigitalOffice.NewsService.Models.Dto.Models;
+using LT.DigitalOffice.NewsService.Models.Dto.Requests;
 using LT.DigitalOffice.NewsService.Validation.Interfaces;
 using MassTransit;
+using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.Logging;
-using System;
-using System.Collections.Generic;
-using System.Linq;
 
 namespace LT.DigitalOffice.NewsService.Business
 {
-    public class CreateNewsCommand : ICreateNewsCommand
+  public class CreateNewsCommand : ICreateNewsCommand
+  {
+    private readonly INewsRepository _repository;
+    private readonly IDbNewsMapper _mapper;
+    private readonly INewsValidator _validator;
+    private readonly IAccessValidator _accessValidator;
+    private readonly IRequestClient<ICheckDepartmentsExistence> _rcCheckDepartmentsExistence;
+    private readonly ILogger<CreateNewsCommand> _logger;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    private List<Guid> CheckDepartmentExistence(Guid? departmentId, List<string> errors)
     {
-        private readonly INewsRepository _repository;
-        private readonly IDbNewsMapper _mapper;
-        private readonly INewsValidator _validator;
-        private readonly IAccessValidator _accessValidator;
-        private readonly IRequestClient<ICheckDepartmentsExistence> _rcCheckDepartmentsExistence;
-        private readonly ILogger<CreateNewsCommand> _logger;
-
-        private List<Guid> CheckDepartmentExistence(Guid? departmentId, List<string> errors)
+      if (!departmentId.HasValue)
+      {
+        return null;
+      }
+      string errorMessage = "Failed to check the existing department.";
+      string logMessage = "Department with id: {id} not found.";
+
+      try
+      {
+        var response = _rcCheckDepartmentsExistence.GetResponse<IOperationResult<ICheckDepartmentsExistence>>(
+          ICheckDepartmentsExistence.CreateObj(new List<Guid> { departmentId.Value })).Result;
+        if (response.Message.IsSuccess)
```

Comment:
you can return Result.Message to not write response.Message every time

Reply:
Work with Department it's not my task, i just use edditorconfig, will can fixed this in other task